Add full KMS migration instructions (#401)
digital-asset/docs.daml.com: docs/2.7.0/docs/canton/usermanual/security.rst

bootstrap) or for an existing deployment.
When the KMS is enabled after a node has been running, the keys are (a) encrypted and stored in this encrypted form
in the Canton node's database, or (b) transparently replaced by external KMS keys. For

> scenario (a) this process is done transparently, while in (b) it :ref:`could be more involved <live_provider_migration>`

Reviewer: ```suggestion
scenario (a) this process is done transparently, while in (b) d :ref:`a node needs to be migrated <live_provider_migration>`
```

Finally, we need to initialize our :ref:`domain <manually-init-domain>` and
:ref:`participants <manually-init-participant>` using the previously registered keys.

.. _live_provider_migration:

> Live Provider Migration

Reviewer: ```suggestion
Participant Node Migration to KMS Crypto Provider
```

Reviewer: "live" suggests that this can be done online without disruption, which is not true.

^^^^^^^^^^^^^^^^^^^^^^^

> To migrate a live participant node connected to a domain with a non KMS-compatible provider

Reviewer: ```suggestion
To migrate an existing participant node connected to a domain with a non KMS-compatible provider
```

and start using KMS external keys we need to manually execute the following steps.
The general idea is to replicate our old node into a new one that uses a KMS provider and connects to a KMS-compatible

> domain (e.g. running JCE with KMS supported encryption and signing keys).

Reviewer: provide a link on how to configure a KMS compatible domain?

Finally, we need to initialize our :ref:`domain <manually-init-domain>` and
:ref:`participants <manually-init-participant>` using the previously registered keys.

.. _participant_kms_migration:

Participant Node Migration to KMS Crypto Provider
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

To migrate an existing participant node connected to a domain with a non KMS-compatible provider
and start using KMS external keys, we need to manually execute the following steps.

> The general idea is to replicate our old node into a :ref:`new one that uses a KMS provider and connects to

Reviewer 2: ```suggestion
The general idea is to replicate the old node into a :ref:`new one that uses a KMS provider and connects to
```

   :language: scala
   :start-after: user-manual-entry-begin: KmsSetupNamespaceDelegation
   :end-before: user-manual-entry-end: KmsSetupNamespaceDelegation
   :dedent:

Secondly, we must recreate all parties of the old participant in the new participant:

.. literalinclude:: /canton/includes/mirrored/enterprise/app/src/test/scala/com/digitalasset/canton/integration/tests/security/AwsKmsCryptoIntegrationTestBase.scala
   :language: scala
   :start-after: user-manual-entry-begin: KmsSetupNamespaceDelegation
   :end-before: user-manual-entry-end: KmsSetupNamespaceDelegation
   :dedent:

Finally, we need to transfer the active contracts of all the parties from the old participant to the new one and
connect to the new domain:

.. literalinclude:: /canton/includes/mirrored/enterprise/app/src/test/scala/com/digitalasset/canton/integration/tests/security/AwsKmsCryptoIntegrationTestBase.scala
   :language: scala
   :start-after: user-manual-entry-begin: KmsMigrateACSofParties
   :end-before: user-manual-entry-end: KmsMigrateACSofParties
   :dedent:

> After all these steps, we have a new participant node with its keys stored and managed by a KMS connected to a domain

Reviewer 2: ```suggestion
The end result is a new participant node with its keys stored and managed by a KMS connected to a domain
```
